### PR TITLE
fix 168

### LINKED ISSUE #168: sysinfo error when rendering html
trying to run any benchmark with current master leads to this error:

```
Writing results to outputs/benchmark_linear_ica_benchopt_run_2021-05-30_21h30m34.html
Traceback (most recent call last):
  File "/Users/alex/miniconda3/bin/benchopt", line 33, in <module>
    sys.exit(load_entry_point('benchopt', 'console_scripts', 'benchopt')())
  File "/Users/alex/miniconda3/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/alex/miniconda3/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/alex/miniconda3/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/alex/miniconda3/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/alex/miniconda3/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/alex/work/src/benchOpt/benchopt/cli/main.py", line 97, in run
    run_benchmark(
  File "/Users/alex/work/src/benchOpt/benchopt/runner.py", line 496, in run_benchmark
    plot_benchmark(save_file, benchmark)
  File "/Users/alex/work/src/benchOpt/benchopt/plotting/__init__.py", line 43, in plot_benchmark
    plot_benchmark_html(fname, benchmark, kinds, display)
  File "/Users/alex/work/src/benchOpt/benchopt/plotting/generate_html.py", line 427, in plot_benchmark_html
    rendered = render_benchmark(
  File "/Users/alex/work/src/benchOpt/benchopt/plotting/generate_html.py", line 301, in render_benchmark
    return Template(
  File "/Users/alex/miniconda3/lib/python3.8/site-packages/mako/template.py", line 473, in render
    return runtime._render(self, self.callable_, args, data)
  File "/Users/alex/miniconda3/lib/python3.8/site-packages/mako/runtime.py", line 878, in _render
    _render_context(
  File "/Users/alex/miniconda3/lib/python3.8/site-packages/mako/runtime.py", line 920, in _render_context
    _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
  File "/Users/alex/miniconda3/lib/python3.8/site-packages/mako/runtime.py", line 947, in _exec_template
    callable_(context, *args, **kwargs)
  File "_Users_alex_work_src_benchOpt_benchopt_plotting_html_templates_benchmark_mako_html", line 38, in render_body
KeyError: 'sysinfo'
```

the function _fetch_cached_run_list seems to remove the sysinfo from the results dictionaries and the rendering fails.

@tanglef do I miss something ? It works for you?

### PR DESCRIPTION
**Pull request by tanglef**
Fix #168 
Because tests passed and `results` webpages were good I forgot to test `sysinfo` in HTML page for singles.
Thank you @agramfort  for finding it out (I'll be more careful with that).

### CODE REVIEW

**Comment by agramfort** [comment]
Thx @tanglef 

Any way we can test this ?

**Comment by agramfort** [comment]
@tomMoral do you see why this was not tested?

**Comment by tanglef** [comment]
If we remove the `--no-plot` in the function that tests the run then the plots are made and it would have returned a `sysinfo` key error

**Comment by agramfort** [comment]
can you do it so we have a non-regression test?

>

**Comment by tanglef** [comment]
It's done

**Comment by agramfort** [comment]
you have CI failures now @tanglef

**Comment by tanglef** [comment]
Yes, but it's just the Julia install which is unstable

**Comment by agramfort** [comment]
can you restart the build so we don't merge something red?

>

**Comment by tanglef** [comment]
All green now :slightly_smiling_face:

File: benchopt/tests/test_cli.py [hunk 1]
```diff
@@ -67,7 +67,7 @@ def test_benchopt_run(self):
         with CaptureRunOutput() as out:
             run([str(DUMMY_BENCHMARK_PATH), '-l', '-d', SELECT_ONE_SIMULATED,
                  '-f', SELECT_ONE_PGD, '-n', '1', '-r', '1', '-p', '0.1',
-                 '--no-plot'], 'benchopt', standalone_mode=False)
+                 ], 'benchopt', standalone_mode=False)
```
Comment (tomMoral) [inline benchopt/tests/test_cli.py:70]: If you do this, you need to clean up after the test :)
Thus here we need to get from the output the name of the HTML page, and then remove it if it exists.

See how it is done for `csv` in `CaptureRunOutput`.

**Review by tomMoral — CHANGES_REQUESTED** [review]
Thx a lot @tanglef .
This looks good but we need to add clean up for the generated file. see comment inline.

File: benchopt/tests/__init__.py [hunk 2]
```diff
@@ -53,7 +53,13 @@ def __exit__(self, exc_class, value, traceback):
...
         if len(self.result_files) >= 1:
             for result_file in self.result_files:
-                Path(result_file).unlink()
+                path_ = Path(result_file)
+                path_.unlink()  # remove csv file
+                dir_ = path_.parents[0]
+                stem = path_.stem
+                html_files = dir_.glob('*' + stem + '*.html')
+                for html_file in html_files:
+                    html_file.unlink(missing_ok=True)  # remove html files
```
Comment (tomMoral) [inline benchopt/tests/__init__.py]: Some variable renaming, but this is completely nitpicking :sweat_smile: 

```suggestion
                result_path = Path(result_file)
                result_path.unlink()  # remove csv file
                result_dir = result_path.parents[0]
                stem = result_path.stem
                for html_file in result_dir.glob(f'*{stem}*.html'):
                     # remove html files associated with this results
                    html_file.unlink(missing_ok=True)
```

**Review by tomMoral — APPROVED** [review]
LGTM! Thanks a lot @tanglef

File: benchopt/tests/__init__.py [hunk 3]
```diff
@@ -59,7 +59,8 @@ def __exit__(self, exc_class, value, traceback):
                 stem = path_.stem
                 html_files = dir_.glob('*' + stem + '*.html')
                 for html_file in html_files:
-                    html_file.unlink(missing_ok=True)  # remove html files
+                    if html_file.exists():
+                        html_file.unlink()  # remove html files
```
Comment (tomMoral) [inline benchopt/tests/__init__.py:63]: How can it not exists as it is given by `glob`?

Comment (tanglef) [reply benchopt/tests/__init__.py:63]: True.... I'll remove this

**Comment by tomMoral** [comment]
Thx @tanglef for the quick fix!

**Pull request merged**